Source/Serilog.Exceptions/Destructurers/SocketExceptionDestructurer.cs

```diff
@@ -0,0 +1,88 @@
+                    { SocketError.ProtocolFamilyNotSupported, "The protocol family is not implemented or has not been configured." },
+                    { SocketError.ProtocolNotSupported, "The protocol is not implemented or has not been configured." },
+                    { SocketError.ProtocolOption, "An unknown, invalid, or unsupported option or level was used with a Socket." },
+                    { SocketError.ProtocolType, "The protocol type is incorrect for this Socket." },
+                    { SocketError.Shutdown, "A request to send or receive data was disallowed because the Socket has already been closed." },
+                    { SocketError.SocketError, "An unspecified Socket error has occurred." },
+                    { SocketError.SocketNotSupported, "The support for the specified socket type does not exist in this address family." },
+                    { SocketError.Success, "The Socket operation succeeded." },
+                    { SocketError.SystemNotReady, "The network subsystem is unavailable." },
+                    { SocketError.TimedOut, "The connection attempt timed out, or the connected host has failed to respond." },
+                    { SocketError.TooManyOpenSockets, "There are too many open sockets in the underlying socket provider." },
+                    { SocketError.TryAgain, "The name of the host could not be resolved. Try again later." },
+                    { SocketError.TypeNotFound, "The specified class was not found." },
+                    { SocketError.VersionNotSupported, "The version of the underlying socket provider is out of range." },
+                    { SocketError.WouldBlock, "An operation on a nonblocking socket cannot be completed immediately." },
+                };
+
+        /// <inheritdoc cref="IExceptionDestructurer.TargetTypes"/>
+        public override Type[] TargetTypes => new[]
+        {
+            typeof(SocketException),
+        };
+
+        /// <inheritdoc cref="IExceptionDestructurer.Destructure"/>
+        public override void Destructure(
+            Exception exception,
+            IExceptionPropertiesBag propertiesBag,
+            Func<Exception, IReadOnlyDictionary<string, object>> destructureException)
+        {
+            base.Destructure(exception, propertiesBag, destructureException);
+
+#pragma warning disable CA1062 // Validate arguments of public methods
+            var socketException = (SocketException)exception;
+            _ = SocketErrorDocumentationBySocketError.TryGetValue(socketException.SocketErrorCode, out var documentation);
+            propertiesBag.AddProperty(nameof(SocketException.SocketErrorCode), socketException.SocketErrorCode);
+            propertiesBag.AddProperty(nameof(SocketException.SocketErrorCode) + "Documentation", documentation ?? "Unknown SocketErrorCode value " + socketException.SocketErrorCode);
```

Comment:
How about naming this `SocketErrorMessage`? I don't mind this either way.

Reply:
When [targeting .NET Core 3.1](https://docs.microsoft.com/en-us/dotnet/api/system.net.sockets.socketexception?view=netcore-3.1), there are three properties on `SocketException`:
1. `ErrorCode`,
2. `NativeErrorCode`, and
3. `SocketErrorCode`.

Presumably they should be consistent and I have only ever seen them be consistent in practice, but the documentation doesn't say this.

Out of an abundance of caution for the possibility that they might not be consistent, I want this additional property that we are enriching to clearly be related to `SocketErrorCode` and not to either of the other two.  Certainly your suggestion of `SocketErrorMessage` goes a long way in that direction because exactly two of these four keys contain the word "Socket".  However, I think the association is even clearer when one is a full prefix of the other.

I also think having the key include a full prefix of the other helps to communicate both that it is not defined on `SocketException` (because if someone checked the documentation, this property would appear next to `SocketErrorCode` since the properties are ordered alphabetically) and it is (somehow) exclusively derived from `SocketErrorCode` since it is not defined on `SocketException`.

In the commit I just pushed, I changed the name from `SocketErrorCodeDocumentation` to `SocketErrorCodeMessage`.

This is your library though, so if you would prefer another name, then I will use that name.